### PR TITLE
Update repo to modernized standards

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #95, #96

User Story: https://dev.azure.com/itsals/E_Retain_Content/_boards/board/t/Auro%20Design%20System/Stories/?workitem=891684

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

* Rebuild and update repo to support modernized standards
* Fix examples to make them work on the docsite
* Add CSS parts

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [x] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ branches:
         # Required. Require branches to be up to date before merging.
         strict: true
         # Required. The list of status checks to require in order to merge into this branch
-        contexts: ["test (11.x)", "test (12.x)", "test (14.x)", "license/cla"]
+        contexts: ["test (18.x)", "license/cla"]
```
Comment (blackfalcon) [inline .github/settings.yml:58]: Compare to https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/.github/settings.yml

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,136 +0,0 @@
-# Semantic Release Automated Changelog
```
Comment (blackfalcon) [inline CHANGELOG.md:1]: Bring back the whole changelog from the previous repo version.

File: docTemplates/README.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
+<!--
```
Comment (blackfalcon) [inline docTemplates/README.md:1]: This file is generated upon request. Should not be committed to version control. Please update the .gitignore https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/.gitignore.temp

File: package.json [hunk 4]
```diff
@@ -18,109 +17,84 @@
...
     "copyfiles": "^2.4.1",
-    "core-js": "^3.19.1",
-    "eslint": "^7.32.0",
-    "eslint-plugin-lit": "^1.6.1",
-    "focus-visible": "^5.2.0",
-    "husky": "^4.3.8",
+    "core-js": "^3.31.0",
+    "eslint": "^8.43.0",
+    "eslint-plugin-jsdoc": "^46.2.6",
+    "husky": "^8.0.0",
```
Comment (blackfalcon) [inline package.json:52]: v8.0.3 is released. Please ensure that all dependencies are at the lastest version possible.

File: src/style.css [hunk 5]
```diff
@@ -0,0 +1,484 @@
+:root{
```
Comment (blackfalcon) [inline src/style.css:1]: None of this should be committed to version control?

File: src/auro-checkbox.scss [hunk 6]
```diff
@@ -3,13 +3,28 @@
...
+   selector-class-pattern,
+   rule-empty-line-before,
+   scss/selector-no-redundant-nesting-selector,
+   no-descending-specificity,
+   no-duplicate-selectors,
+   declaration-empty-line-before */
+
 // Import Auro tokens
 @import "./../node_modules/@alaskaairux/design-tokens/dist/tokens/SCSSVariables";
 @import "./../node_modules/@alaskaairux/design-tokens/dist/tokens/SassCustomProperties";
```
Comment (blackfalcon) [inline src/auro-checkbox.scss:23]: This is no longer needed since we are not post-processing any of the CSS custom properties.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Also update the .npmignore file https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/.npmignore.temp

File: .github/settings.yml [hunk 7]
```diff
@@ -55,7 +55,7 @@ branches:
         # Required. Require branches to be up to date before merging.
         strict: true
         # Required. The list of status checks to require in order to merge into this branch
-        contexts: ["test (11.x)", "test (12.x)", "test (14.x)", "license/cla"]
+        contexts: ["test ([abstractNodeVersion])", "license/cla"]
```
Comment (blackfalcon) [inline .github/settings.yml:58]: `[abstractNodeVersion]` is a template variable. This needs to be updated with a version of Node to test. E.g. `18.x`.

File: docs/upstream.md [hunk 8]
```diff
@@ -1,6 +1,6 @@
 # Maintaining your fork's upstream relationship
 
-So you forked this repo. That's awesome. But don't forget you keep your `master` branch in sync with the upstream `master` branch to ensure that your pull requests will always be able to be merged.
+So you forked this repo. That's awesome. But don't forget you keep your `main` branch in sync with the upstream `main` branch to ensure that your pull requests will always be able to be merged.
```
Comment (blackfalcon) [inline docs/upstream.md:3]: This repo still uses a `master` branch. Unless we do the work to rename the branch.

File: docs/upstream.md [hunk 9]
```diff
@@ -42,23 +42,28 @@ upstream        https://github.com/AlaskaAirlines/auro-checkbox.git (push)
 
 #### Sync upstream
 
-To sync your master brach with your upstream master branch, there are two ways you can do this.
-
-One way is to fetch and merge. This will work if your `master` branch shares the same history as the upstream brach sans merged updates since fork or last sync.
+To sync your `main` branch with the upstream `main` branch, there are two ways you can do this. The preferred method is to fetch and rebase. Please do not merge down.
```
Comment (blackfalcon) [inline docs/upstream.md:45]: This repo still uses a `master` branch. Unless we do the work to rename the branch.

File: docs/upstream.md [hunk 10]
```diff
@@ -42,23 +42,28 @@ upstream        https://github.com/AlaskaAirlines/auro-checkbox.git (push)
...
 
 ```bash
-$ git checkout master
+$ git checkout main
```
Comment (blackfalcon) [inline docs/upstream.md:48]: This repo still uses a `master` branch. Unless we do the work to rename the branch.

File: demo/demo.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+<div class="exampleWrapper">
+  <custom-header display="display">Salutations World!</custom-header>
+</div>
+<auro-accordion lowProfile justifyRight>
+  <span slot="trigger">See code</span>
+  ```html
+  <custom-header display="display">Salutations World!</custom-header>
+  ```
+
+</auro-accordion>
```
Comment (blackfalcon) [inline demo/demo.md:151]: this example is incorrect.

File: package.json [hunk 12]
```diff
@@ -18,109 +17,84 @@
...
-    "@polymer/iron-demo-helpers": "^3.1.0",
-    "@polymer/iron-icons": "^3.0.1",
-    "@polymer/polymer": "^3.3.1",
-    "@semantic-release/changelog": "^5.0.1",
-    "@semantic-release/git": "^9.0.0",
-    "@semantic-release/npm": "^7.1.3",
-    "@webcomponents/webcomponentsjs": "^2.6.0",
-    "autoprefixer": "^10.4.0",
-    "babel-loader": "^8.2.3",
+    "@alaskaairux/design-tokens": "^3.14.0",
```
Comment (blackfalcon) [inline package.json:32]:
```
 Patch Update Backwards-compatible bug fixes.
❯◯ typescript devDep  5.1.3  ❯  5.1.5  https://www.typescriptlang.org/

 Minor Update New backwards-compatible features.
 ◯ eslint-plugin-jsdoc devDep  46.3.0  ❯  46.4.0  https://github.com/gajus/eslint-plugin-jsdoc#readme
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: docs/upstream.md [hunk 13]
```diff
@@ -64,12 +69,12 @@ $ git reset --hard upstream/master
...
+$ git checkout master
 
 $ git fetch upstream
 $ git merge upstream/master
 ```
 
 ## Maintain upstream
 
-Once you have connected your local forked clone to the upstream repo, maintenance is the key. It is important to ensure that your `master` branch is always updated BEFORE you create a new branch intended to be used with a new pull request. This will help to eliminate any potential issues with merging when the `master` branches are out of sync.
+Once you have connected your local forked clone to the upstream repo, mastertenance is the key. It is important to ensure that your `master` branch is always updated BEFORE you create a new branch intended to be used with a new pull request. This will help to eliminate any potential issues with merging when the `master` branches are out of sync.
```
Comment (blackfalcon) [inline docs/upstream.md:80]: Please be careful with search and replace. You replaced `maintenance` with `mastertenance`.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: apiExamples/basic.html [hunk 14]
```diff
@@ -0,0 +1,7 @@
+<auro-checkbox-group>
+  <span slot="legend">Form label goes here</span>
+  <auro-checkbox value="checkbox option" name="example1">Checkbox option</auro-checkbox>
+  <auro-checkbox value="checkbox option" name="example2" checked>Checkbox option</auro-checkbox>
+  <auro-checkbox value="checkbox option" name="example3">Checkbox option</auro-checkbox>
+  <auro-checkbox value="checkbox option" name="example4">Checkbox option</auro-checkbox>
+</auro-checkbox-group>
```
Comment (blackfalcon) [inline apiExamples/basic.html:7]: None of these examples have an ID on the `auro-checkbox` node. None of these examples pass accessibility tests. 

Screen Shot 2023-07-01 at 8 19 34 PM

**Review by blackfalcon — DISMISSED** [review]

File: apiExamples/basic.html (hunk 14 above)
Comment (jordanjones243) [reply apiExamples/basic.html:7]: What is this tool? I would like to test the accessibility myself as well.

Comment (jason-capsule42) [reply apiExamples/basic.html]: @blackfalcon what is the a11y guideline for ID attributes on checkboxes?

**Review by jason-capsule42 — APPROVED** [review]

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Please update the API descriptions. See https://auro.alaskaair.com/components/auro/checkbox/api for content to preserve. 

The "Required" example should be on the main demo page, this is a core feature of the group element. 

Missing the "Horizontal Checkbox Group" and the "Horizontal Checkbox Group limitation" examples.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Missing updates

1. ISSUE TEMPLATES
2. Labels in the settings.yml

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**